pkg/backend/openshiftcluster/create_test.go

```diff
@@ -0,0 +1,58 @@
+package openshiftcluster
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+
+	"github.com/jim-minter/rp/pkg/api"
+	"github.com/jim-minter/rp/pkg/util/mocks"
+	"github.com/jim-minter/rp/pkg/util/mocks/mock_azureclient/mock_resources"
+	"github.com/jim-minter/rp/pkg/util/mocks/mock_database"
+	"github.com/jim-minter/rp/pkg/util/mocks/mock_env"
+	"github.com/jim-minter/rp/pkg/util/mocks/mock_env/shared/mock_dns"
+	"github.com/jim-minter/rp/pkg/util/mocks/mock_install"
+	"github.com/jim-minter/rp/pkg/util/mocks/mock_subnet"
+)
+
+func TestManagerCreate(t *testing.T) {
+	ctx := context.TODO()
+	gmc := gomock.NewController(t)
+	defer gmc.Finish()
+	logger := logrus.NewEntry(logrus.StandardLogger())
+
+	doc := &api.OpenShiftClusterDocument{
+		Key:              "foo",
+		OpenShiftCluster: mocks.MockOpenShiftCluster(),
+	}
+
+	mDB := mock_database.NewMockOpenShiftClusters(gmc)
+	mDB.EXPECT().Patch(gomock.Any(), gomock.Any()).Return(doc, nil)
```

Comment:
`Manager#Create` seem to mainly consist of calls of other functions which ideally should be tested separately (like `subnet.Split` and db's `Patch`, `validation.ValidateInstallConfig` from installer, etc).

Hovever there are few pieces that potentially need testing in `Manager#Create`:
* Document patch callback where we populate properties `SSHKey`, `StorageSuffix`, `DomainName` if they are not present.
* Constructed `installConfig` and `platformCreds`

But we do not cover them here: we replace them with `gomock.Any()`